berlekamp_welch.go

```diff
@@ -106,15 +116,30 @@ func (fc *FEC) Correct(shares []Share) error {
 			}
 			data, err := fc.berlekampWelch(shares, j)
 			if err != nil {
-				return err
+				return nil, err
 			}
 			for _, share := range shares {
+				if share.Data[j] != data[share.Number] && !badShareMap[share.Number] {
+					badShares = append(badShares, share)
```

Comment:
I think it would be enough to return just the list of bad share numbers, not the complete bad shares.